### PR TITLE
Prioritize lower-compute-unit-limit transactions

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
#28751. Promote transactions with lower requested computer units.

#### Summary of Changes
Update `Ord` for `ImmutableDeserializedPacket` to prioritize lower compute-unit-limit packets *after* priority.

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
This doesn't strictly bump txs with requested CU, but those with **lower** CU limits. I think in most cases this will result in the same thing, however does not if a tx actually needs higher CUs than the default of 200k.

File: core/src/immutable_deserialized_packet.rs [hunk 1]
```diff
@@ -129,8 +129,12 @@ impl PartialOrd for ImmutableDeserializedPacket {
 }
 
 impl Ord for ImmutableDeserializedPacket {
+    /// Order by priority (higher first), then by compute_unit_limit (lower first)
     fn cmp(&self, other: &Self) -> Ordering {
-        self.priority().cmp(&other.priority())
+        match self.priority().cmp(&other.priority()) {
+            Ordering::Equal => other.compute_unit_limit().cmp(&self.compute_unit_limit()),
```
Comment (taozhu-chicago) [inline core/src/immutable_deserialized_packet.rs:135]: Not fully convinced if it's a good idea to prioritize txs requested less CUs. Maybe better to incentivize user to adapt requested CU with economic means. But if we really want to using prioritization as tool to encourage user to requested CU, then maybe only by a boolean flag of requested vs non-requested. (perhaps better move the discussion to issue itself)

Comment (segfaultdoc) [reply core/src/immutable_deserialized_packet.rs:135]: Does reversing the comparison result in better block packing?

**Comment by apfitzge** [comment]
See comments above, not the direction we want to go.

**Pull request closed without merging**